ledger/consumer/smartcontract/tx.go

```diff
@@ -1,20 +1,35 @@
 package smartcontract
 
 import (
+	"encoding/binary"
+	fmt "fmt"
+	"hash"
+
 	"github.com/golang/protobuf/proto"
 	"github.com/golang/protobuf/ptypes/any"
 	"go.dedis.ch/fabric/crypto"
+	"go.dedis.ch/fabric/crypto/common"
 	"go.dedis.ch/fabric/encoding"
+	"go.dedis.ch/fabric/ledger/arc"
 	"go.dedis.ch/fabric/ledger/consumer"
-	"go.dedis.ch/fabric/ledger/inventory"
 	"golang.org/x/xerrors"
 )
 
+type action interface {
+	encoding.Packable
+
+	hashTo(hash.Hash, encoding.ProtoMarshaler) error
+}
+
 // transaction is an atomic execution.
 //
 // - implements ledger.transaction
 type transaction struct {
-	hash []byte
+	hash      []byte
+	nonce     uint64
+	action    action
+	identity  crypto.PublicKey
```

Comment:
Yes because it doesn't make sense to have multiple signature for one transaction. We can later implement a deferred transaction smart contract to have multiple signatures but when creating a transaction, you usually have one signing key.